drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerPresenter.java

```diff
@@ -413,18 +413,47 @@ public void execute(Collection<String> availableParentRuleNames) {
         if (isGridPinned()) {
             view.getGridLayerView().flipToGridWidget(dtPresenter.getView());
         }
+
+        handlePermissions(dtPresenter);
+    }
+
+    void handlePermissions(final GuidedDecisionTableView.Presenter dtPresenter) {
+        if (dtPresenter.hasEditableColumns()) {
+            view.enableColumnOperationsMenu();
+        }
     }
 
     void refreshDefinitionsPanel(final GuidedDecisionTableView.Presenter dtPresenter) {
         final GuidedDecisionTable52 model = dtPresenter.getModel();
-        view.setEnableColumnCreation(dtPresenter.getAccess().isEditable());
+
+        view.setEnableColumnCreation(isColumnCreationEnabled(dtPresenter));
         view.refreshAttributeWidget(model.getAttributeCols());
         view.refreshMetaDataWidget(model.getMetadataCols());
         view.refreshConditionsWidget(model.getConditions());
         view.refreshActionsWidget(model.getActionCols());
         view.refreshColumnsNote(dtPresenter.hasColumnDefinitions());
     }
 
+    Boolean isColumnCreationEnabled(final GuidedDecisionTableView.Presenter dtPresenter) {
+
+        final Boolean decisionTableIsEditable = !dtPresenter.isReadOnly();
+        final Boolean decisionTableHasEditableColumns = dtPresenter.hasEditableColumns();
+
+        return decisionTableHasEditableColumns && decisionTableIsEditable;
+    }
+
+    @Override
+    public boolean isColumnCreationEnabledToActiveDecisionTable() {
+
+        final Optional<GuidedDecisionTableView.Presenter> decisionTable = Optional.ofNullable(getActiveDecisionTable());
```

Comment:
I'm sure you could do something more funky with ```Optional``` (than check ``isPresent()```) but I'm old fashioned and am happy with this :-) Just adding a note as someone (younger!) somewhere might say something!

Reply:
;-)